src/ruby/puppetserver-lib/puppet/server/http_response.rb

```diff
@@ -0,0 +1,52 @@
+require 'puppet/http/response'
+
+class Puppet::Server::HttpResponse < Puppet::HTTP::Response
+  attr_reader :url
+
+  def initialize(java_response, url)
+    @java_response = java_response
+    @url = url
+  end
+
+  def code
+    @java_response.get_status
+  end
+
+  def reason
+    @java_response.get_reason_phrase
+  end
+
+  def body
+    @java_response.get_body
+  end
+
+  def read_body
+    yield body
+  end
+
+  # The HTTP standard counts all codes 200-299 as successes
+  def success?
+    code > 199 && code < 300
+  end
+
+  def [](name)
+    @java_response.get_headers[name]
+  end
+
+  # Yield each header name and value. Returns an enumerator if no block is given.
+  def each_header
+    if block_given?
+      @java_response.get_headers.each do |name, value|
+        yield(name, value)
+      end
+    else
+      @java_response.get_headers.each
+    end
```

Comment:
Is there a more concise way to do this? Little rusty on my Ruby block handling.

Reply:
I think the `&block` param is always considered optional and you can then pass it to `each` and it will do what you want, ie:
```rb
def foo(&block)
   {a: 1, b: 2, c: 3}.each(&block)
end

foo
=> #<Enumerator: {:a=>1, :b=>2, :c=>3}:each>

foo {|k,v| puts v }
1
2
3
```

Reply:
Oh great, thanks!

Reply:
Oh still need to fix this.